lib/dirac_improved_staggered.cpp

```diff
@@ -127,6 +127,37 @@ namespace quda {
   {
     // do nothing
   }
+  
+  void DiracImprovedStaggered::SmearOp(ColorSpinorField &out, const ColorSpinorField &in, const double &, const double &,
```

Comment:
is this an exact code duplication from `dirac_staggered.cpp` ? Just wondering if we can avoid that.

Reply:
They are identical. To avoid duplication, I think we should introduce an intermediate class between the Dirac class and them (DiracStaggered and DiracImprovedStaggered). Or, if we remove (or replace) checkSpinorAlias() defined in the Dirac class, it would also be possible to create a shared function outside the class (or in another mother class).

Reply:
I think we can leave this as it is for now and consider cleanup / consolidation after merging.